From ad51c9152b1784bc7fbee5d52775fc974b8a49ea Mon Sep 17 00:00:00 2001
From: Jacek Pielaszkiewicz <j.pielaszkie@samsung.com>
Date: Wed, 15 Jan 2014 15:07:54 +0100
Subject: [PATCH 1083/1302] Split uImage on "clean" uImage and Device Tree
 blob.

1. The change is due to upgrade u-boot to latest upstream version
   which support separate uImage and DTB.
   Addtionally current defult u-boot configuration for trats2 assumes
   that kernel image and DTB are delivered as separat files.

2. DTB files are located in /boot directory.

3. Together with uImage are delivered all DTBs which are compatible with
   tizen_defconfig kernel configuration.

Change-Id: I60a581edb4edccbf2ef704de525409da6572c4f8
Signed-off-by: Jacek Pielaszkiewicz <j.pielaszkie@samsung.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 packaging/linux-kernel.spec | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/packaging/linux-kernel.spec b/packaging/linux-kernel.spec
index 935e62f..211efd8 100644
--- a/packaging/linux-kernel.spec
+++ b/packaging/linux-kernel.spec
@@ -58,10 +58,8 @@ make EXTRAVERSION="-%{build_id}" %{config_name}
 make EXTRAVERSION="-%{build_id}" %{?_smp_mflags}
 
 # 2. Build uImage
-make EXTRAVERSION="-%{build_id}" zImage %{?_smp_mflags}
+make EXTRAVERSION="-%{build_id}" uImage %{?_smp_mflags}
 make EXTRAVERSION="-%{build_id}" dtbs %{?_smp_mflags}
-cat arch/arm/boot/zImage arch/arm/boot/dts/%{defaultDtb}  > bImage
-mkimage -A arm -C none -O linux -a 40008000 -e 40008000 -n 'Linux 3.10 Tizen kernel' -d bImage uImage
 
 # 3. Build modules
 make EXTRAVERSION="-%{build_id}" modules %{?_smp_mflags}
@@ -78,7 +76,10 @@ mkdir -p %{buildroot}/lib/modules/%{fullVersion}
 mkdir -p %{buildroot}/boot/
 
 # 2. Install uImage, System.map, ...
-install -m 755 uImage %{buildroot}/boot/
+install -m 755 arch/arm/boot/uImage %{buildroot}/boot/
+install -m 644 arch/arm/boot/dts/*.dtb %{buildroot}/boot/
+mv %{buildroot}/boot/exynos4412-m0.dtb %{buildroot}/boot/exynos4412-trats2.dtb
+
 install -m 644 System.map %{buildroot}/boot/System.map-%{fullVersion}
 install -m 644 .config %{buildroot}/boot/config-%{fullVersion}
 
@@ -138,6 +139,7 @@ rm -rf %{buildroot}
 %files
 %license COPYING
 /boot/uImage
+/boot/*.dtb
 /boot/System.map*
 /boot/config*
 /lib/modules/%{fullVersion}/kernel
-- 
1.8.3.2

